runtime/src/bank.rs

```diff
@@ -7608,12 +7608,13 @@ impl Bank {
     pub fn verify_snapshot_bank(
         &self,
         test_hash_calculation: bool,
-        accounts_db_skip_shrink: bool,
+        skip_shrink: bool,
+        force_clean: bool,
         last_full_snapshot_slot: Slot,
         base: Option<(Slot, /*capitalization*/ u64)>,
     ) -> bool {
         let (_, clean_time_us) = measure_us!({
-            let should_clean = !accounts_db_skip_shrink && self.slot() > 0;
+            let should_clean = force_clean || (!skip_shrink && self.slot() > 0);
```

Comment:
IIRC, self.slot() > 0 was there previously before the skipping was added. We may still want it. Maybe it protects against some pathological case in tests or test clusters? Or maybe the tests don't exist anymore or clean is happy with slot==0 now?

Reply:
I think you can leave it alone. I did want another set of eyes to look at this and think about it. Should only affect tests or test networks. mnb, testnet, etc. will always be slot > 0.